tests/check/check_test.go

```diff
@@ -53,9 +54,12 @@ func testCheck(t *testing.T, engine string) {
 	)
 	require.NoError(t, err)
 	defer conn.Close()
+	t.Run("testCheck", func(t *testing.T) {
```

Comment:
i'm curious why we need this here?

Reply:
need to ensure all the child test are done before tearing down connection

Reply:
https://pkg.go.dev/testing#T.Run

> Run runs f as a subtest of t called name. It runs f in a separate goroutine and **blocks until f returns or calls t.Parallel to become a parallel test.** Run reports whether f succeeded (or at least did not fail before calling t.Parallel). Run may be called simultaneously from multiple goroutines, but all such calls must return before the outer test function for t returns.

I'm not sure if this will be enough 🤔

Reply:
We didn't see premature tear down of connection so far.

Reply:
Yesterday when I attempted something similar (not your change but very similar), I saw that error.

https://go.dev/play/p/MsVMZ5mrqPs

Reply:
Ok, yeah, this works! https://go.dev/blog/subtests

```go
func TestTeardownParallel(t *testing.T) {
    // <setup code>
    // This Run will not return until its parallel subtests complete.
    t.Run("group", func(t *testing.T) {
        t.Run("Test1", parallelTest1)
        t.Run("Test2", parallelTest2)
        t.Run("Test3", parallelTest3)
    })
    // <tear-down code>
}
```